tiatoolbox/tools/registration/wsi_registration.py

```diff
@@ -536,6 +531,356 @@ def estimate_affine_transform(
+            moving_matched_points = np.delete(
+                moving_matched_points, duplicate_ind, axis=0
+            )
+            quality = np.delete(quality, duplicate_ind)
+
+        return fixed_matched_points, moving_matched_points, quality
+
+    def register(
+        self,
+        fixed_img: np.ndarray,
+        moving_img: np.ndarray,
+        fixed_mask: np.ndarray,
+        moving_mask: np.ndarray,
+        transform_initializer: np.ndarray = None,
+    ) -> np.ndarray:
+        """Image Registration.
+
+        This function aligns a pair of images using Deep
+        Feature based Registration (DFBR) method.
+
+        Args:
+            fixed_img (:class:`numpy.ndarray`):
+                A fixed image.
+            moving_img (:class:`numpy.ndarray`):
+                A moving image.
+            fixed_mask (:class:`numpy.ndarray`):
+                A binary tissue mask for the fixed image.
+            moving_mask (:class:`numpy.ndarray`):
+                A binary tissue mask for the moving image.
+            transform_initializer (:class:`numpy.ndarray`):
+                A rigid transformation matrix.
+
+        Returns:
+            :class:`numpy.ndarray`:
+                An affine transformation matrix.
+
+        """
+        if len(fixed_img.shape) != 3 or len(moving_img.shape) != 3:
+            raise ValueError(
+                "The required shape for fixed and moving images is n x m x 3."
+            )
+
+        if fixed_img.shape[2] != 3 or moving_img.shape[2] != 3:
+            raise ValueError("The input images are expected to have 3 channels.")
+
+        if not transform_initializer:
+            transform_initializer = prealignment(
+                fixed_img[:, :, 0], moving_img[:, :, 0], fixed_mask, moving_mask
+            )
+
+        moving_img = cv2.warpAffine(
+            moving_img, transform_initializer[0:-1][:], fixed_img.shape[:2][::-1]
+        )
+        moving_mask = cv2.warpAffine(
+            moving_mask, transform_initializer[0:-1][:], fixed_img.shape[:2][::-1]
+        )
+
+        (
+            fixed_tissue,
+            fixed_mask,
+            moving_tissue,
+            moving_mask,
+            tissue_top_left_coord,
+        ) = self.get_tissue_regions(fixed_img, fixed_mask, moving_img, moving_mask)
+        features = self.extract_features(fixed_tissue, moving_tissue)
+        fixed_matched_points, moving_matched_points, quality = self.feature_mapping(
+            features
+        )
+
+        (
+            fixed_matched_points,
+            moving_matched_points,
+            quality,
+        ) = self.filtering_matching_points(
+            fixed_mask,
+            moving_mask,
+            fixed_matched_points,
+            moving_matched_points,
+            quality,
+        )
+
+        tissue_transform = DFBRegister.estimate_affine_transform(
+            fixed_matched_points, moving_matched_points
+        )
+
+        moving_tissue = cv2.warpAffine(
```

Comment:
```suggestion
        # Apply Affine transformation
        moving_tissue = cv2.warpAffine(
```

Reply:
https://github.com/TissueImageAnalytics/tiatoolbox/blob/a040932dda0d4641e69f41d2b19b7f85c16e7ca7/tiatoolbox/tools/registration/wsi_registration.py#L808-L811

Reply:
Removed Affine from the comment as I was getting a spelling error in pre-commit. 
https://github.com/TissueImageAnalytics/tiatoolbox/blob/695a97a48e399b7d19d646ab8d7a6322c8356e13/tiatoolbox/tools/registration/wsi_registration.py#L808-L811